File: src/main/kotlin/com/miro/miroappoauth/dto/LineBorderStyle.java
package com.miro.miroappoauth.dto;

import org.jetbrains.annotations.Nullable;

import java.util.stream.Stream;

public enum LineBorderStyle {
    normal,
    dashed,
    dotted;

    public static LineBorderStyle fromString(@Nullable String value) {
        return Stream.of(values())
                .filter(cnst -> cnst.name().equalsIgnoreCase(value))
                .findFirst()
                .orElse(normal);
    }
}

File: src/main/kotlin/com/miro/miroappoauth/client/v1/LineBorderStyle.kt
package com.miro.miroappoauth.client.v1

import java.util.stream.Stream

enum class LineBorderStyle {
    normal, dashed, dotted;

    companion object {
        fun fromString(value: String?): LineBorderStyle {
            return Stream.of(*values())
                .filter { cnst: LineBorderStyle -> cnst.name.equals(value, ignoreCase = true) }
                .findFirst()
                .orElse(normal)
        }
    }
}
